### DISCUSSION TITLE
Assistance Required for MycilaJSY Library Issue with ESP32 and JSY-MK-333

### DISCUSSION THREAD

**Discussion by owaistech** [mathieucarbou/MycilaJSY#11] [General]
Dear Mathieu,

I hope you are doing well. I am working on a project using your MycilaJSY library to interface with the JSY-MK-333 module on an ESP32 WROOM microcontroller. I greatly appreciate your efforts in developing this library and providing support for JSY modules.

However, I encountered an issue while attempting to read parameters from the JSY-MK-333. Specifically, I am receiving the following error during compilation:

/Users/awais./Downloads/MycilaJSY-main 2/examples/Read/Read.ino: In function 'void loop()':  
/Users/awais./Downloads/MycilaJSY-main 2/examples/Read/Read.ino:45:9: error: 'class Mycila::JSY' has no member named 'toJson'  
   45 |     jsy.toJson(doc.to<JsonObject>());  
      |         ^~~~~~  
exit status 1  
Compilation error: 'class Mycila::JSY' has no member named 'toJson'

I was attempting to use the toJson() function to retrieve all available parameters from the JSY-MK-333 in JSON format. The example code in your library references this function, but it appears to be missing or unsupported in the current library version.

Details of My Setup:

	1.	Microcontroller: ESP32 WROOM
	2.	Library Version: Latest 
	3.	Objective: Retrieve and display all available parameters (e.g., voltage, current, power, etc.) from the JSY-MK-333 module.

Questions:

	1.	Is the toJson() function supported in the current version of the MycilaJSY library?
	2.	If not, what is the recommended method to retrieve all parameters for the JSY-MK-333 using your library?
	3.	Are there alternative functions or approaches we should use to achieve this functionality?

I would greatly appreciate your guidance on resolving this issue and implementing a solution to retrieve the parameters efficiently. Please let me know if you need additional details about the setup or error logs.

**Comment by mathieucarbou** [comment]
Hello, please read the doc: https://github.com/mathieucarbou/MycilaJSY?tab=readme-ov-file#json-support (and look for example at the PIO file).
